Remove "All Woo Stores powered by WordPress.com" copy in non-NUX screens

chihsuan opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Close https://github.com/Automattic/wp-calypso/issues/87443

It's confusing to see "All Woo Stores powered by WordPress.com" copy in non-NUX screens as it's not relevant to the user. This PR removes the copy from non-NUX screens. We use `wccomFrom` to determine if the user is on a NUX screen or not.
## Proposed Changes

* Remove All Woo stores are powered by WordPress.com! when not from NUX context

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

**NUX flow**

1.  Log out from woo.com
2. Go to https://woo.com/start/#/installation
3. Click on "Try Woo Express for free"
4. Replace URL host with your local calypso or calypso live host
5. Observe there's `All Woo stores are powered by WordPress.com!`
6. Click on "Log in"
7.  Observe there's `All Woo stores are powered by WordPress.com!`

<img width="675" alt="Screenshot 2024-02-16 at 12 02 39" src="https://github.com/Automattic/wp-calypso/assets/4344253/d83182e8-0765-479a-aeb4-a5caba5ea473">

<img width="1344" alt="Screenshot 2024-02-16 at 11 58 25" src="https://github.com/Automattic/wp-calypso/assets/4344253/1ce78c6d-d9f6-41cd-8857-34b61174b193">


**non-NUX flow**

1.  Log out from woo.com
2. Go to https://woo.com/sso?next=%2F
3. Click on "Try Woo Express for free"
4. Replace URL host with your local calypso or calypso live host
5. Observe there's no `All Woo stores are powered by WordPress.com!`
6. Click on "Log in"
8.  Observe there's no `All Woo stores are powered by WordPress.com!`

<img width="644" alt="Screenshot 2024-02-16 at 12 04 40" src="https://github.com/Automattic/wp-calypso/assets/4344253/899ff294-42ef-4f30-9efa-2b90be0338bd">

<img width="1350" alt="Screenshot 2024-02-16 at 11 47 44" src="https://github.com/Automattic/wp-calypso/assets/4344253/96147894-a65c-463d-b820-566ff3ec3892">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

chihsuan committed:
Remove "All Woo Stores powered by WordPress.com" copy in non-NUX screens

rjchow reviewed (commented):
There's a small bit of history behind this -
I introduced some of that wording as a stopgap while maintaining the intent of informing the user why they are logging into a wpcom account when they're trying to use a woo.com site. On reflection it does seem a bit misleading, but we might want to also consider the original intent.

I don't have too strong of an opinion either way as it seems neither the current nor the proposed solutions have an edge over the other. Maybe we could trouble @lauroraa (sorry, tagged the wrong Laura originally) 😀

ilyasfoo reviewed (changes requested):
Tests well! A minor thing though, it looks like there is a quick flash of change when clicking either "login" or "sign up", where `All Woo Express stores are powered by WordPress.com!'` disappears and reappears when coming from NUX:

https://github.com/Automattic/wp-calypso/assets/3747241/fb5361f3-fbe5-44a5-8b9b-a64043627ea4

chihsuan committed:
Get wccomFrom from query params, instead of parsing the oauth_redirect_url

To make it consistent with login flow, and void flash content when the user is switching from the signup flow to the login flow.

chihsuan committed:
Add get-wccom-from selector and use it in login and signup

This selector is used to get the wccom-from value from the state.
We use it in login and signup to get the wccom-from value

ilyasfoo reviewed (approved):
Awesome, thanks @chihsuan! Tested well 🚀

rjchow reviewed (approved):
I think Laura has a lot on her plate right now so let's go with this for now

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11383037

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @chihsuan for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11383037).
